src/settings.cpp

```diff
@@ -171,19 +171,24 @@ const uint16 INIFILE_VERSION = (IniFileVersion)(IFV_MAX_VERSION - 1); ///< Curre
  * @param str the current value of the setting for which a value needs found
  * @param len length of the string
  * @param many full domain of values the ONEofMANY setting can have
+ * @param many_cnvt callback procedure when loading value mechanism fails
  * @return the integer index of the full-list, or -1 if not found
  */
-size_t OneOfManySettingDesc::ParseSingleValue(const char *str, size_t len, const std::vector<std::string> &many)
+size_t OneOfManySettingDesc::ParseSingleValue(const char *str, size_t len, const std::vector<std::string> &many, OnConvert *many_cnvt)
 {
-	/* check if it's an integer */
-	if (isdigit(*str)) return std::strtoul(str, nullptr, 0);
-
 	size_t idx = 0;
 	for (auto one : many) {
 		if (one.size() == len && strncmp(one.c_str(), str, len) == 0) return idx;
 		idx++;
 	}
 
+	/* We failed our lookup; try the conversion function next. */
+	idx = many_cnvt(str);
+	if (idx != (size_t)-1) return idx;
```

Comment:
```suggestion
	if (many_cnvt != nullptr) {
		idx = many_cnvt(str);
		if (idx != (size_t)-1) return idx;
	}
```
As far as I can see `many_cnvt` can be `nullptr`.